lightningd/pay.c

```diff
@@ -1006,6 +1054,46 @@ static void json_sendpay(struct command *cmd,
 		return;
 	}
 
+	/* Default (realm = 0) realm / hop_data, and other route data */
+	custom_route_data = route_to_custom_route_data(cmd->ld, route);
```

Comment:
pass custom_route_data as a pointer to route_to_custom_route_data here instead of returning.